src/menu_put_utils.c

```diff
@@ -0,0 +1,101 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   menu_put_utils.c                                   :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gwolf <gwolf@student.42vienna.com>         +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/12/27 15:16:08 by gwolf             #+#    #+#             */
+/*   Updated: 2024/01/06 13:43:08 by gwolf            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "menu_put.h"
+
+t_vec2i	ft_put_3d_point(t_putinfo put, t_vec3 point)
+{
+	t_numinfo	numinfo;
+	const int	num_x = put.pos.x + 30;
+
+	numinfo.pad = 5;
+	numinfo.prec = 2;
+	ft_put_str(put, " x:");
+	numinfo.numf = point.x;
+	ft_put_float((t_putinfo){.mlx_ptrs = put.mlx_ptrs,
```

Comment:
The following lines are very similar except for the coordinates. Could be a separate function which is called 3x with the individual coordinates.

Reply:
Good point - Refactored:
- new ft_put_coord() outputs a single coordinate
- ft_put_3d_point(), ft_put_unit_vec() and ft_put_colour() use this ft
- they also get passed a name to print (e.g. Centre or Normal)